modules/tensor_mechanics/src/materials/ADIsotropicPlasticityStressUpdate.C

```diff
@@ -0,0 +1,238 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADIsotropicPlasticityStressUpdate.h"
+
+#include "Function.h"
+#include "ElasticityTensorTools.h"
+
+registerADMooseObject("TensorMechanicsApp", ADIsotropicPlasticityStressUpdate);
+
+defineADValidParams(
+    ADIsotropicPlasticityStressUpdate,
+    ADRadialReturnStressUpdate,
+    params.addClassDescription("This class uses the discrete material in a radial return isotropic "
+                               "plasticity model.  This class is one of the basic radial return "
+                               "constitutive models, yet it can be used in conjunction with other "
+                               "creep and plasticity materials for more complex simulations.");
+    // Linear strain hardening parameters
+    params.addParam<FunctionName>("yield_stress_function",
+                                  "Yield stress as a function of temperature");
+    params.addParam<Real>("yield_stress",
+                          0.0,
+                          "The point at which plastic strain begins accumulating");
+    params.addParam<FunctionName>("hardening_function",
+                                  "True stress as a function of plastic strain");
+    params.addParam<Real>("hardening_constant", 0.0, "Hardening slope");
+    params.addCoupledVar("temperature", 0.0, "Coupled Temperature");
+    params.addDeprecatedParam<std::string>(
+        "plastic_prepend",
+        "",
+        "String that is prepended to the plastic_strain Material Property",
+        "This has been replaced by the 'base_name' parameter");
+    params.set<std::string>("effective_inelastic_strain_name") = "effective_plastic_strain";);
+
+template <ComputeStage compute_stage>
+ADIsotropicPlasticityStressUpdate<compute_stage>::ADIsotropicPlasticityStressUpdate(
+    const InputParameters & parameters)
+  : ADRadialReturnStressUpdate<compute_stage>(parameters),
+    _plastic_prepend(adGetParam<std::string>("plastic_prepend")),
+    _yield_stress_function(
+        isParamValid("yield_stress_function") ? &getFunction("yield_stress_function") : NULL),
+    _yield_stress(adGetParam<Real>("yield_stress")),
+    _hardening_constant(adGetParam<Real>("hardening_constant")),
+    _hardening_function(isParamValid("hardening_function") ? &getFunction("hardening_function")
+                                                           : NULL),
+    _yield_condition(-1.0), // set to a non-physical value to catch uninitalized yield condition
```

Comment:
I seem to remember you correcting me on not using something like this but some libmesh variable...can't be for sure though...

Reply:
I probably told you to use `libMesh::invalid_uint`, but this is a floating point number.

Reply:
ah, good to know, thanks